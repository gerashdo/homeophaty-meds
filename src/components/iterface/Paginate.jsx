import { PaginateItem } from "./PaginateItem"
import { usePaginate } from "../../hooks/usePaginate"

import './paginate.css'


export const Paginate = ({ onClickItem, totalPages, currentPage: actualPage }) => {

    const { setCurrentPage, currentPage, pages } = usePaginate( totalPages, actualPage )

    const onItemClick = ( number ) => {
        setCurrentPage( number )
        onClickItem( number )
    }

    return (
        <div className="paginate-container">
            {
                pages.map( page => (
                    <PaginateItem 
                        key={ page }
                        number={ page }
                        active={ currentPage === page ? true : false }
                        onItemClick={ onItemClick }
                    />
                ))
            }
        </div>
    )
}
