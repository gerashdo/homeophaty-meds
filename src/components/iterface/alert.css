
.alert-container{
    align-items: center;
    background-color: var(--color-primary-darker);
    border-radius: .75rem;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
    color: white;
    display: flex;
    gap: 1rem;
    position: fixed;
    top: 50px;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 1.5rem 3rem 1.5rem 1.5rem;
    overflow: hidden;
}

.alert-container span {
    font-size: 1.5rem;
}

.alert-container.success {
    background-color: var(--color-secondary);
}

.alert-container.error {
    background-color: var(--color-negative);
}

.alert-container .close {
    position: absolute;
    top: 5px;
    right: 10px;
    border-radius: .25rem;
    cursor: pointer;
    font-size: var(--font-normal);
    padding: 0.125rem 0.5rem;
    transition: all 0.2s ease;
    opacity: 0.7;
}

.alert-container .close:hover {
    opacity: 1;
}