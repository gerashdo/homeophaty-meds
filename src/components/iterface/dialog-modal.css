
dialog{
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

dialog.dialog-modal{
    border: 0;
    border-radius: .75rem;
    /* display: flex; */
    /* flex-direction: column; */
    /* gap: 1rem; */
    padding: 1.5rem 2rem;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
}

dialog.dialog-modal::backdrop{
    background: rgba(0, 0, 0, 0.3);
}

dialog.dialog-modal .title h4{
    font-weight: var(--font-bold);
    font-size: var(--font-size-subtitle);
    text-align: start;
    margin-bottom: 1rem;
}

dialog.dialog-modal .content {
    margin-bottom: 2rem;
}

dialog.dialog-modal .footer{
    display: flex;
    justify-content: flex-end;
    gap: 1rem;
}