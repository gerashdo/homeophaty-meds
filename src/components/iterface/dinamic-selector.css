
.dinamic-selector{
    background-color: var(--color-gray-background);
    border-radius: 0 0px 10px 10px;
    box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.2);
    position: absolute;
    width: 100%;
    max-height: 200px;
    overflow: auto;
    top: 40px;
}

.dinamic-selector ul {
    list-style: none;
}

.dinamic-selector ul li {
    padding: 0.5rem 1rem;
    cursor: pointer;
}

.dinamic-selector ul li:hover {
    background-color: var(--color-primary-lighter);
    color: white;
}