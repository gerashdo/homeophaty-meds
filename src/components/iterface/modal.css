dialog{
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

dialog.modal{
    border: 0;
    border-radius: .75rem;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
}

dialog.modal .modal-content{
    margin: 3rem;
}

dialog.modal::backdrop{
    background: rgba(0, 0, 0, 0.3);
}

dialog.modal .modal-close{
    position: absolute;
    top: 1rem;
    right: 1rem;
    color: white;
}
