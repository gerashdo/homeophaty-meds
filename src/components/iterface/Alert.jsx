import { useDispatch, useSelector } from "react-redux"
import { BsFillCheckCircleFill } from "react-icons/bs"

import './alert.css'
import { useEffect } from "react"
import { stopAlert } from "../../store/slices/ui"

export const Alert = ({ children }) => {
    const dispatch = useDispatch()
    const { alertActive, alertMessage, alertType } = useSelector( state => state.ui )

    const closeAlert = () => {
        dispatch( stopAlert() )
    }

    useEffect(() => {
        if( alertActive ){
            setTimeout( () => {
               closeAlert()
            }, 8000)
        }
    }, [ alertActive ])

    return (
        <>
        { children }

        {
            alertActive
            ?(<div className={`alert-container ${alertType}`}>
                <span>
                    <BsFillCheckCircleFill />
                </span>
                <p>{ alertMessage }</p>
                <span 
                    className="close"
                    onClick={ closeAlert }
                >x</span>
            </div>)
            : null
        }
        </>
    )
}
