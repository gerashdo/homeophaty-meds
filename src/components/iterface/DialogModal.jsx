
import { useEffect, useRef } from 'react'
import './dialog-modal.css'

export const DialogModal = ({ title, content, open, onClose, onAccept }) => {
  const ref = useRef(null)

  useEffect(() => {
    if (open) {
      ref.current?.showModal()
    } else {
      ref.current?.close()
    }
  }, [open])

  const handleAccept = () => {
    onAccept()
    onClose()
  }

  return (
    <dialog
      ref={ref}
      className='dialog-modal'
    >
      <header className='title'>
        <h4>{title}</h4>
      </header>

      <p
        className='content'
      >
        {content}
      </p>
      <footer
        className='footer'
      >
        <button
          onClick={() => onClose()}
          className='simple'
        >
          Cancelar
        </button>
        <button
          onClick={handleAccept}
          className='primary'
        >
          Aceptar
        </button>
      </footer>
    </dialog>
  )
}
