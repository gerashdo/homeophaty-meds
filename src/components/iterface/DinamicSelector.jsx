
import { useEffect, useState } from 'react'
import { searchStringInMed } from '../../helpers'
import PropTypes from 'prop-types'

import './dinamic-selector.css'

export const DinamicSelector = ({ resetValues, valueForFilter, onSetValue }) => {
  const [valuesForUse, setValuesForUse] = useState(resetValues)

  useEffect(() => {
    if (valueForFilter) {
      setValuesForUse(searchStringInMed(valueForFilter, resetValues))
    } else {
      setValuesForUse(resetValues)
    }
  }, [valueForFilter, resetValues])

  const setSelectedValue = ({ target }) => {
    onSetValue(target.innerText)
  }

  return (
    <div
      className='absolute bg-white rounded-md shadow z-20 w-72 inset-x-0 max-h-60 top-16'
    >
      <ul>
        {
          valuesForUse.length > 0
            ? valuesForUse.map(med => (
              <li
                key={med._id}
                onMouseDown={setSelectedValue}
                className='hover:bg-gray-200 cursor-pointer p-2'
              >
                {med.name} {med.ch}
              </li>
            ))
            : <li data-testid='no-results'>Sin resultados</li>
        }
      </ul>
    </div>
  )
}

DinamicSelector.propTypes = {
  resetValues: PropTypes.array.isRequired,
  valueForFilter: PropTypes.string,
  onSetValue: PropTypes.func.isRequired
}
