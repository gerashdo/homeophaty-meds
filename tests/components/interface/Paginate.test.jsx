import { fireEvent, render, screen } from "@testing-library/react";
import { Paginate } from "../../../src/components/iterface/Paginate";


describe('test for <Paginate /> component', () => {
    const onClickItem = jest.fn()
    const totalPages = 10

    test('should render the component properly', () => {
        const currentPage = 1

        const { container } = render( <Paginate 
            onClickItem={ onClickItem }
            totalPages={ totalPages }
            currentPage={ currentPage }
        /> )

        expect( container ).toMatchSnapshot()
    });

    test('should call the onClickItem function if a item is clicked', () => {
        const currentPage = 1

        render( <Paginate 
            onClickItem={ onClickItem }
            totalPages={ totalPages }
            currentPage={ currentPage }
        /> )

        fireEvent.click( screen.getAllByTestId('item-container')[1] )

        expect( onClickItem ).toHaveBeenCalled()
    });
});