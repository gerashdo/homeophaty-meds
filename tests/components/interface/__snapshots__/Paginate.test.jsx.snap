// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <Paginate /> component should render the component properly 1`] = `
<div>
  <div
    class="paginate-container"
  >
    <div
      class="paginate-item active"
      data-testid="item-container"
    >
      1
    </div>
    <div
      class="paginate-item "
      data-testid="item-container"
    >
      2
    </div>
    <div
      class="paginate-item "
      data-testid="item-container"
    >
      3
    </div>
    <div
      class="paginate-item "
      data-testid="item-container"
    >
      4
    </div>
  </div>
</div>
`;
